pyrovision/inference/pyronear_predictor.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) Pyronear contributors.
+# This file is dual licensed under the terms of the CeCILL-2.1 and AGPLv3 licenses.
+# See the LICENSE file in the root of this repository for complete details.
+
+import torch
+from torch import nn
+from torchvision import transforms
+from pyrovision.models.utils import cnn_model
+import holocron
+from holocron.models.utils import load_pretrained_params
+import configparser
+
+
+def read_config_file(filename):
+    """Read the config file and return it as a dictionary."""
+    config = configparser.ConfigParser()
+    config.read(filename)
+    conf = {}
+    for parts in config:
+        for key in config[parts]:
+            conf[key] = config[parts][key]
+
+    return conf
+
+
+class PyronearPredictor:
+    """Class used the last pyronear model to predict if a sigle frame contain a fire or not.
+
+    Parameters
+    ----------
+    configFile: str
+        Path to the configuration file
+
+    checkpointPath: str,
+        Path to model checkpoint, if you give an path here it will overwrite the one in the config file
+
+    """
+
+    def __init__(self, configFile, checkpointPath=None):
+        """Init Pyronear Predictor."""
+        # Load config file
+        self.config = read_config_file(configFile)
+        if checkpointPath:
+            self.config['url'] = checkpointPath
+
+        # Model definition
+        self.model = self.get_model()
+
+        # Transform definition
+        self.transforms = self.get_transforms()
+
+        self.sigmoid = nn.Sigmoid()
+
+    def get_transforms(self):
+        """Define Transforms definition."""
+        size = int(self.config['image_size'])
+        normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+
+        if self.config['use_center_crop']:
+            tf = transforms.Compose([transforms.Resize(size=(size)),
+                                     transforms.CenterCrop(size=size),
+                                     transforms.ToTensor(),
+                                     normalize
+                                     ])
+        else:
+            tf = transforms.Compose([transforms.Resize(size=(size)),
+                                     transforms.ToTensor(),
+                                     normalize
+                                     ])
```

Comment:
Each user can have his/user own transforms, so perhaps we can let them choose. Similarly to https://github.com/frgfm/Holocron/blob/master/references/classification/README.md, if we specify on which dataset it was pretrained and with which image size, they can write that by themselves

Reply:
As I said in the previous comment the goal is to create a unique predictor. This will only be changed when we get a better model.

Reply:
But I can remove the if statement, and just leave the transform I need

Reply:
cf. my previous reply
To sum up:
- inference information would be better stored as model attributes
- in pyronear-engine, we use the attributes (normalization parameters for instance) to create the pre-processing correctly and process outputs

Another possibility is to embed the preprocessing in the model, like in https://github.com/pytorch/vision/blob/master/torchvision/models/detection/faster_rcnn.py#L233 where there is a `transforms` module in the model that process the input tensor before passing it to the core model.